### ISSUE TITLE
First row scrolls out of view on the initial mount

### ISSUE THREAD

**Issue by turisap** [fullcalendar/fullcalendar#6507]
### Reduced Test Case

[REDUCED TEST CASE](https://codesandbox.io/s/wrong-scroll-on-mount-8tfby?file=/src/calendar.js).

### Bug Description

Conditions:
* we have more than 50 rows in the resource timeline
* we have an event in the very first row
* we use `dateSet` prop

For some reasons,  if aforementioned conditions are met, the timeline component scrolls the first row out of view on the initial mount. As a workaround I used a `ref` to keep track of the initial mount and to set dates only after it happens, as follows:

```
   const setDates = (dates: DatesSetArg) => {
        if(isInit.current) {
            setDateChosen(dates);
        }

        if(!isInit.current) {
            isInit.current = true;
        }
    };
```

### Screenshots
<img width="734" alt="Screenshot 2021-08-10 at 13 59 33" src="https://user-images.githubusercontent.com/28143822/128857004-3cb3c9e6-1025-42eb-885f-e9fe1426cb24.png">

**Comment by acerix** [comment]
I couldn't replicate the issue [without React](https://codepen.io/acerix/pen/wvdRVPW?editors=001), so I believe this is specific to the React component.

Since `datesSet` is called when the calendar is initialised, having it update the state which causes the calendar to immediately initialise a second time seems to mess up the scrolling calculation.  So that can be avoided by not initialising a new calendar on state change.

It seems related to https://github.com/fullcalendar/fullcalendar/issues/5923 , I found that using a non-numeric resource ID also avoids the issue:

https://codesandbox.io/s/wrong-scroll-on-mount-forked-t9l7r?file=/src/calendar.js

**Comment by turisap** [comment]
@acerix, I have just checked [your codesandbox](https://codesandbox.io/s/wrong-scroll-on-mount-forked-t9l7r?file=/src/calendar.js) with space-prefixed resource IDs. I might be missing something, but it feels like it is also necessary to prefix `resourceID`s for events. If that is done, the problem persists. If not, `resourceID`s on events and IDs on resources are no longer in match and there is no events shown in the timeline (as in your example).

**Comment by acerix** [comment]
You are correct, changing the ID only worked because it was getting rid of the event.  That hack worked in a similar case and seemed to work here because I forgot the event was part of the issue.

**Comment by arshaw** [comment]
[Happening for Angular](https://github.com/fullcalendar/fullcalendar-angular/issues/413) as well. These connectors sometimes add content after the updateSize computation has finished. Probably the root of the problem..
